### ISSUE TITLE
[feature request] Make Before, After, BeforeSuite, AfterSuite data driven

### ISSUE THREAD

**Issue by AdamOakman** [Codeception/CodeceptJS#1405]
#### What are you trying to achieve?

Data driven hooks: Before After BeforeSuite AfterSuite

For example it would be nice to make use of something like this:

```
Data(datatable).Before((I, current) => {
	I.amOnPage('/');
        I.loginAs(current.user);
});
```

#### What do you get instead?

Nothing...

### Details

* CodeceptJS version: 2.0.1

Any chance we can expect this update soon?

**Comment by DavertMik** [comment]
To be honest this is a very strange feature and yeah, I'm not sure it can be released soon.
Why do you need that in hooks, you can still use this in tests

**Comment by AdamOakman** [comment]
This would give an ability to shorten the body of multiple test cases inside single feature. Logging into an application is a basic prerequisite operation which requires multiple sets of data such as usernames and passwords depending on permissions granted to particular user. I would like to move "log in" step inside single Before hook so it is not necessary to use each time in all the test cases. Let me give you an example:

Now:
`Feature('Check admin permissions');

Before((I) => {
	I.amOnPage('/');
});

Data(datatable).Scenario('TC 001 check something', (I, current) => {	
	I.loginAs(current.user);
	// steps to perform
});

Data(datatable).Scenario('TC 002 check something', (I, current) => {
	I.loginAs(current.user);
	// steps to perform
});

Data(datatable).Scenario('TC 003 check something', (I, current) => {
	I.loginAs(current.user);
	// steps to perform
});
...`

My suggestion:
`Feature('Check admin permissions');

Data(datatable).Before((I, current) => {
	I.amOnPage('/');
    I.loginAs(current.user);
});

Scenario('TC 001 check something', (I) => {	
	// steps to perform
});

Scenario('TC 002 check something', (I) => {	
	// steps to perform
});

Scenario('TC 003 check something', (I) => {	
	// steps to perform
});

...`

This would give hundreds of other use cases in my opinion and would make the test code look more clean. Especially when one feature contains many testcases.

**Comment by DavertMik** [comment]
This problem will be solved in 2.0.3 with autoLogin plugin. 
Stay tuned!
